app/src/main/java/ch/sdp/vibester/activity/profile/ProfileActivity.kt

```diff
@@ -1,204 +1,226 @@
+import com.google.zxing.qrcode.QRCodeWriter
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel
+import dagger.hilt.android.AndroidEntryPoint
+import net.glxn.qrgen.android.MatrixToImageWriter
+import javax.inject.Inject
+
+/**
+ * Display user profile's data (image, username, scores, etc.) in UI
+ */
+@AndroidEntryPoint
+open class ProfileActivity : AppCompatActivity(), OnItemClickListener {
+    @Inject
+    lateinit var dataGetter: DataGetter
+
+    @Inject
+    lateinit var authenticator: FireBaseAuthenticator
+
+    @Inject
+    lateinit var imageGetter: ImageGetter
+
+    private val imageSize = 1000
+    val imageRequestCode = 100
+
+    private var followings: MutableList<User> ? = null
+    private var profileFriendsAdapter: ProfileFollowingAdapter?= null
+
+    /**
+     * Generic onCreate method belonging to ProfileActivity.
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        requestWindowFeature(Window.FEATURE_NO_TITLE)
+        super.onCreate(savedInstanceState)
+        supportActionBar?.hide()
+        setContentView(R.layout.activity_profile)
+
+        setupRecycleViewForFriends()
+        followings = ArrayList()
+
+        setRetToMainBtnListener()
+        setFollowingScoresBtnListener(R.id.profile_scores, R.id.profile_scroll_stat, R.id.profile_scroll_following)
+        setFollowingScoresBtnListener(R.id.profile_following, R.id.profile_scroll_following, R.id.profile_scroll_stat)
+
+        queryDatabase()
+    }
+
+    private fun setupRecycleViewForFriends() {
+        findViewById<RecyclerView>(R.id.profile_followingList).apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = followings?.let { ProfileFollowingAdapter(it, this@ProfileActivity) }
+            setHasFixedSize(true)
+        }
+    }
+
+    private fun showFriendsPosition(friends: MutableList<User>?) {
+        profileFriendsAdapter = ProfileFollowingAdapter(friends!!, this)
+        findViewById<RecyclerView>(R.id.profile_followingList)!!.adapter = profileFriendsAdapter
+    }
+
+
+    //check the UID here not sure
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK && requestCode == imageRequestCode) {
+            imageGetter.uploadFile("profileImg/${dataGetter.getCurrentUser()?.uid}", data?.data!!) {
+                imageGetter.fetchImage("profileImg/${dataGetter.getCurrentUser()?.uid}", this::setImage)
+            }
+        }
+    }
+
+
+    /**
+     * Generic listener for the return to main button.
+     */
+    private fun setRetToMainBtnListener() {
+        findViewById<FloatingActionButton>(R.id.profile_returnToMain).setOnClickListener {
+            IntentSwitcher.switch(this, MainActivity::class.java)
+            finish()
+        }
+    }
+
+    private fun setFollowingScoresBtnListener(btnId: Int, show: Int, hide: Int) {
```

Comment:
For consistency reasons maybe add a Javadoc for the private function even if this is not necessary. But all the others in the same file have javadoc.